src/reconcile/rec_row.c

```diff
@@ -464,7 +465,12 @@ __wti_rec_row_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
     WT_INTL_FOREACH_END;
 
     /* Write the remnant page. */
-    return (__wti_rec_split_finish(session, r));
+    ret = __wti_rec_split_finish(session, r);
+    /*
+     * Set the ref_changes state to zero if there were no concurrent changes while reconciling the
+     * internal page.
+     */
+    __wt_atomic_casv16(&ref->ref_changes, 0, 0);
```

Comment:
We should read the value at the start of reconciling this ref. And cas it at the end.
__wt_atomic_casv16(&ref->ref_changes, prev_count, 0);